extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeJavaScriptExtractorTest.java

```diff
@@ -19,6 +19,16 @@ public void setup() throws IOException {
         NewPipe.init(DownloaderTestImpl.getInstance());
     }
 
+    @Test
+    public void testExtractJavaScriptUrlIframe() throws ParsingException {
+        YoutubeJavaScriptExtractor.extractJavaScriptUrl();
+    }
+
+    @Test
+    public void testExtractJavaScriptUrlEmbed() throws ParsingException {
+        YoutubeJavaScriptExtractor.extractJavaScriptUrl("d4IGg5dqeO8");
+    }
```

Comment:
Can you add some assertions? As it stands, I could delete the whole implementation of the methods and these test would still pass.

Otherwise remove the tests/ undo you last commit.

Reply:
Oh, I thought it was fine since it would throw an exception in case it can't find an URL. Is it still necessary to add some assertions?

Reply:
But what about the cases where it finds an url, but they don't lead to the player js file?

Reply:
Resolved! I've made it check whether it ends with `base.js`, there are a lot of URL formats for the player URL, and hence I avoided using a regex.